rocks/RocksProperties.java

```diff
@@ -0,0 +1,146 @@
+import org.rocksdb.RocksDBException;
+import org.rocksdb.TableProperties;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.function.Function;
+
+import static com.vaticle.typedb.core.common.collection.Bytes.MB;
+import static com.vaticle.typedb.core.common.exception.ErrorMessage.Internal.STORAGE_PROPERTY_EXCEPTION;
+
+/**
+ * A list of interesting properties to retrieve from RocksDB.
+ * All properties are visible here:
+ * https://github.com/facebook/rocksdb/blob/20357988345b02efcef303bc274089111507e160/include/rocksdb/db.h#L750
+ */
+public enum RocksProperties {
+
+    ACTIVE_MEM_TABLE_SIZE("rocksdb.cur-size-active-mem-table", (size) ->
+            String.format("%.2f mb", Double.parseDouble(size) / MB)),
+    ALL_MEM_TABLES_SIZE("rocksdb.size-all-mem-tables", (size) ->
+            String.format("%.2f mb", Double.parseDouble(size) / MB)),
+    BLOCK_CACHE_PINNED_USAGE("rocksdb.block-cache-pinned-usage", (size) ->
+            String.format("%.2f mb", Double.parseDouble(size) / MB)),
+    BLOCK_CACHE_CAPACITY("rocksdb.block-cache-capacity", (size) ->
+            String.format("%.2f mb", Double.parseDouble(size) / MB)),
+    BLOCK_CACHE_USAGE("rocksdb.block-cache-usage", (size) ->
+            String.format("%.2f mb", Double.parseDouble(size) / MB)),
+    SST_READERS_MEMORY_ESTIMATE("rocksdb.estimate-table-readers-mem", (size) ->
+            String.format("%.2f mb", Double.parseDouble(size) / MB)),
+    ACTIVE_SNAPSHOTS("rocksdb.num-snapshots", (num) -> num),
+    OLDEST_ACTIVE_SNAPSHOT_UNIX_SECONDS("rocksdb.oldest-snapshot-time", (oldestTime) ->
+            String.format("%.2f sec", (System.currentTimeMillis() / 1000.0 - Long.parseLong(oldestTime)))),
+    LIVE_VERSIONS("rocksdb.num-live-versions", (num) -> num),
+    SST_FILES_LIVE_SIZE("rocksdb.live-sst-files-size", (size) ->
+            String.format("%.2f mb", Double.parseDouble(size) / MB)),
+    SST_FILES_HELD_SIZE("rocksdb.total-sst-files-size", (size) ->
+            String.format("%.2f mb", Double.parseDouble(size) / MB)),
+    MEM_TABLE_FLUSH_PENDING("rocksdb.mem-table-flush-pending", (size) ->
+            String.format("%.2f mb", Double.parseDouble(size) / MB)),
+    COMPACTION_BYTES_ESTIMATE("rocksdb.estimate-pending-compaction-bytes", (size) ->
+            String.format("%.2f mb", Double.parseDouble(size) / MB)),
+    NUM_RUNNING_COMPACTIONS("rocksdb.num-running-compactions", (num) -> num),
+    NUM_RUNNING_FLUSH("rocksdb.num-running-flushes", (num) -> num);
+
+
+    private final String property;
+    private final Function<String, String> formatter;
+
+    RocksProperties(String property, Function<String, String> formatter) {
+        this.property = property;
+        this.formatter = formatter;
+    }
+
+    private String getFormatted(OptimisticTransactionDB rocksDB) throws RocksDBException {
+        return formatter.apply(rocksDB.getProperty(property));
+    }
+
+    static class Logger implements Runnable {
```

Comment:
we bundle a logger for Properties that does things like formatting and aggregations that can be run once or periodically